core/server/worker/src/main/java/alluxio/worker/block/BlockMetadataManager.java

```diff
@@ -242,6 +242,22 @@ public StorageDir getDir(BlockStoreLocation location) {
     return getTier(location.tierAlias()).getDir(location.dir());
   }
 
+  /**
+   * Get the block store location based on the medium type.
+   *
+   * @param mediumType medium type
+   * @return the blockstorelocation object
+   */
+  public BlockStoreLocation getMedium(String mediumType) {
+    for (StorageTier tier : getTiers()) {
+      for (StorageDir dir : tier.getStorageDirs()) {
+        if (dir.getDirMedium().equals(mediumType)) {
+          return new BlockStoreLocation(tier.getTierAlias(), dir.getDirIndex());
```

Comment:
Won't this always return the first storage dir that matches the medium type (but there could be multiple)?